sql/catalyst/src/test/scala/org/apache/spark/sql/connector/catalog/SupportsPartitionManagementSuite.scala

```diff
@@ -248,7 +266,10 @@ class SupportsPartitionManagementSuite extends SparkFunSuite {
   test("truncatePartition") {
     val table = catalog.loadTable(ident)
     val partTable = new InMemoryPartitionTable(
-      table.name(), table.schema(), table.partitioning(), table.properties())
+      table.name(),
```

Comment:
Would it be better to add a constructor to `InMemoryPartitionTable`? Also, since `InMemoryPartitionTable` is code used for testing, it might be acceptable to directly modify the default constructor definition, but I'm not sure if there are third-party projects that depend on this test code.

Reply:
I think we can create a new one and let the old one throw an exception?